src/GenAPI/Microsoft.DotNet.GenAPI/SyntaxGeneratorExtensions.cs

```diff
@@ -43,5 +64,24 @@ public static SyntaxNode DeclarationExt(this SyntaxGenerator syntaxGenerator, IS
                 throw new ArgumentException(ex.Message, symbol.ToDisplayString());
             }
         }
+
+        private static ConstructorInitializerSyntax GenerateBaseConstructorInitializer(ImmutableArray<IMethodSymbol> baseTypeConstructors)
+        {
+            ConstructorInitializerSyntax constructorInitializer = SyntaxFactory.ConstructorInitializer(SyntaxKind.BaseConstructorInitializer);
+
+            IMethodSymbol baseConstructor = baseTypeConstructors.OrderBy(c => c.Parameters.Count()).First();
+            foreach (IParameterSymbol parameter in baseConstructor.Parameters)
+            {
+                IdentifierNameSyntax identifier;
+                if (parameter.Type.IsValueType)
+                    identifier = SyntaxFactory.IdentifierName("default");
+                else
+                    identifier = SyntaxFactory.IdentifierName("default!");
```

Comment:
Is that what we did in the legacy GenAPI?

Reply:
This comment suggests that there should be a better way to figure out if a parameter type accepts null or not: https://github.com/dotnet/arcade/blob/608f3c7aed68dec1314d75601c6f143747e78cc6/src/Microsoft.Cci.Extensions/Writers/CSharp/CSDeclarationWriter.Methods.cs#L456-L463

cc @eerhardt